core/controllers/editor.py

```diff
@@ -769,6 +770,109 @@ def get(self, exploration_id, escaped_state_name):
         })
 
 
+class FetchIssuesHandler(EditorHandler):
+    """Handler used for retrieving the list of unresolved issues in an
+    exploration.
+    """
+
+    @acl_decorators.can_view_exploration_stats
+    def get(self, exp_id):
+        """Handles GET requests."""
+        exp_version = self.request.get('exp_version')
+        exp_issues = stats_services.get_exp_issues(exp_id, exp_version)
+        if exp_issues is None:
+            raise self.PageNotFoundException(
+                'Invalid exploration ID %s' % (exp_id))
+        unresolved_issues = []
+        for issue in exp_issues.unresolved_issues:
+            if issue.is_valid:
+                unresolved_issues.append(issue)
+        exp_issues.unresolved_issues = unresolved_issues
+        exp_issues_dict = exp_issues.to_dict()
+        self.render_json(exp_issues_dict['unresolved_issues'])
+
+
+class FetchPlaythroughHandler(EditorHandler):
+    """Handler used for retrieving a playthrough."""
+
+    @acl_decorators.can_view_exploration_stats
+    def get(self, unused_exploration_id, playthrough_id):
+        """Handles GET requests."""
+        playthrough = stats_services.get_playthrough_by_id(playthrough_id)
+        if playthrough is None:
+            raise self.PageNotFoundException(
+                'Invalid playthrough ID %s' % (playthrough_id))
+        self.render_json(playthrough.to_dict())
+
+
+class ResolveIssueHandler(EditorHandler):
+    """Handler used for resolving an issue. Currently, when an issue is
+    resolved, the issue is removed from the unresolved issues list in the
+    ExplorationIssuesModel instance, and all corresponding playthrough
+    instances are deleted.
+    """
+
+    def _require_exp_issue_dict_is_valid(self, exp_issue_dict):
```

Comment:
Hmm, I don't like this. This should all be contained in some `ExplorationIssue.from_backend_dict` class method, where it can stay coupled closely to the ExplorationIssue class itself. Put another way, the editor should not be responsible for making sure that exp_issues_dict is valid, the consumer should.

Reply:
@brianrodri I don't think that is actually the point of this. This is more linked to the controller because we are actually checking whether the dict passed in from the front end has the correct keys and values. This practice is used in quite a few controllers actually. 

@seanlip What do you think?

Reply:
It's not unreasonable for the controller to call ExplorationIssue.from_backend_dict(). That satisfies both the requirements and I think it's preferable because when you update the definition of ExplorationIssue you'll (presumably) update the definition of this one too.

Reply:
Sounds right. We can reduce duplication too, if I think about it. Done.